prowler/providers/microsoft365/services/entra/entra_password_hash_sync_enabled/entra_password_hash_sync_enabled.py

```diff
@@ -0,0 +1,45 @@
+from typing import List
+
+from prowler.lib.check.models import Check, CheckReportMicrosoft365
+from prowler.providers.microsoft365.services.entra.entra_client import entra_client
+
+
+class entra_password_hash_sync_enabled(Check):
+    """
+    Check if password hash synchronization is enabled for hybrid Microsoft Entra deployments.
+
+    This check verifies that password hash synchronization is enabled in hybrid Microsoft Entra deployments.
+    Enabling password hash sync ensures that on-premises passwords are synchronized to Microsoft Entra,
+    facilitating seamless authentication and enhancing leaked credential protection. Without password hash
+    synchronization, users might have to manage multiple passwords and detection of leaked credentials would be compromised.
+
+    Note: This control applies only to hybrid deployments using Microsoft Entra Connect sync and does not apply to federated domains.
```

Comment:
Should we check then if the domain is federated to avoid false positives?

Reply:
But you can have multiple domains... 🤔 Not sure how this setting would behave in that case.